agent-control/src/agent_control/config.rs

```diff
@@ -703,6 +781,140 @@ k8s:
         assert!(diff.is_empty());
     }
 
+    ////////////////////////////////////////////////////////////////////////////////////
+    // Tests for TryFrom<OpampRemoteConfig> for AgentControlDynamicConfig
+    ////////////////////////////////////////////////////////////////////////////////////
+
+    #[rstest]
+    #[case::single_config_multiple_agents(
+        r#"{"config1": "{\"agents\": {\"agent-a\": {\"agent_type\": \"foo/bar:0.0.1\"}, \"agent-b\": {\"agent_type\": \"foo/bar:0.0.1\"}}}"}"#,
+        r#"{"agents": {"agent-a": {"agent_type": "foo/bar:0.0.1"}, "agent-b": {"agent_type": "foo/bar:0.0.1"}}}"#
+    )]
+    #[case::multiple_configs_different_agents(
+        r#"{"config1": "{\"agents\": {\"agent-a\": {\"agent_type\": \"foo/bar:0.0.1\"}}}", "config2": "{\"agents\": {\"agent-b\": {\"agent_type\": \"foo/bar:0.0.1\"}}}"}"#,
+        r#"{"agents": {"agent-a": {"agent_type": "foo/bar:0.0.1"}, "agent-b": {"agent_type": "foo/bar:0.0.1"}}}"#
+    )]
+    #[case::config_with_both_versions(
+        r#"{"config1": "{\"agents\": {\"agent-a\": {\"agent_type\": \"foo/bar:0.0.1\"}}, \"chart_version\": \"1\", \"cd_chart_version\": \"2\"}"}"#,
+        r#"{"agents": {"agent-a": {"agent_type": "foo/bar:0.0.1"}}, "chart_version": "1", "cd_chart_version": "2"}"#
+    )]
+    #[case::multiple_configs_split_versions(
+        r#"{"config1": "{\"agents\": {\"agent-a\": {\"agent_type\": \"foo/bar:0.0.1\"}}, \"chart_version\": \"1\"}", "config2": "{\"agents\": {\"agent-b\": {\"agent_type\": \"foo/bar:0.0.1\"}}, \"cd_chart_version\": \"2\"}"}"#,
+        r#"{"agents": {"agent-a": {"agent_type": "foo/bar:0.0.1"}, "agent-b": {"agent_type": "foo/bar:0.0.1"}}, "chart_version": "1", "cd_chart_version": "2"}"#
+    )]
+    #[case::config_with_versions_and_another_with_agent(
+        r#"{"config1": "{\"agents\": {}, \"chart_version\": \"1\", \"cd_chart_version\": \"2\"}", "config2": "{\"agents\": {\"agent-b\": {\"agent_type\": \"foo/bar:0.0.1\"}}}"}"#,
+        r#"{"agents": {"agent-b": {"agent_type": "foo/bar:0.0.1"}}, "chart_version": "1", "cd_chart_version": "2"}"#
+    )]
+    #[case::empty_agents(r#"{"config1": "{\"agents\": {}}"}"#, r#"{"agents": {}}"#)]
```

Comment:
I am not super sold on this format because it is hard to follow to see what is being tested, but it's ok.